### DISCUSSION TITLE
Headset-mode

### DISCUSSION THREAD

**Discussion by dadanielat** [egzumer/uv-k5-firmware-custom#380] [Ideas]
Would it be possible to add a headset-mode? At the moment using a headset is nearly impossible, because of the loud popping noise when a channel is keyed up and the NF driver stage is activated. When enabled, the NF output should never be completely turned off, so you would hear the popping only once, the remaining static noise wouldn't be noticeable. I know this would limit battery life.

**Comment by ErikS-web** [comment]
What type of radio and what revision level of board (look inside) dou you have ?

**Comment by egzumer** [comment]
I myself am not going to add any headset mode, that is for sure.
I have no idea what the NF is. I don't know which popping you are talking about. I know there are cases when that happens, but which one do you mean, I don't understand that part.
Popping sound happens when audio commands are not executed in correct order. I know 1o11 did some corrections a while back, I didn't bother to port those.

**Reply by gartnl** [reply]
Educated guess: dadanielat is German and NF means Niederfrequenz . Better known as AF or audio frequency ..

**Comment by ErikS-web** [comment]
It also might be the "DC on the audio"-problem.
Lets wait for photo of internal of the radio.

**Comment by egzumer** [comment]
@dadanielat Ok, I see what you mean. I didn't think about that before. This is caused by the unfortunate quansheng design. The internal speaker is connected to both sides of the amplifier, so there is no pop. External speaker is only connected to one. This is a more shitty design than I thought.
I'll take a note of that, but I'm currently overloaded with other things.

**Comment by gartnl** [comment]
Many HT's have this problem. By Baofeng UV-B5 is just the same. Many digital shortwave receivers too. They all shut down the ( single-ended used ) AF-amp, whereas older analogue designs just muted the amp at the expense of batterylife. It's not QS specific.

**Comment by ErikS-web** [comment]
Suggest to read #267 and #339 and try to place an elco somewhere.

**Reply by egzumer** [reply]
It has nothing to do with that

**Comment by gartnl** [comment]
@ErikS-web This is an age-old problem with audio amplifiers. A single-ended amp raises the output from zero to mid supply on start. Audible as a 'plop' in the speaker, as current flows through the elco and speaker to ground. A familliar sound if you have used old transistor radio's.
A dual-ended (or bridged or dc-amp) raises both outputs at the same time, so no current flows through the speaker and eliminating the need for a capacitor. High-end audio-amplifiers use this configuration as it has better bass-response and allows more power. Often combined with a delayed relay on the speaker connection to give the amplifier time to settle ( prolonged difference in output voltage could damage the speaker ).
The HT has a modern integrated bridged amp, very accurate on start-up so no relay is needed. 
The internal speaker is connected between the two outputs, but the external speaker only to one end with the other end to ground. This is what causes the DC problem we mitigate with an elco(*). 
The HT shuts down the amp when nothing is received (or no tone is send out ) to save energy. It's outputs go down to zero. As soon as audio is required  the amplifier is turned on, giving the plop in the external speaker.

The best current solution would be to turn 'Beep' and 'Voice' off. At least you want hear a sound when using the keys. The plop will still be there when receiving a signal starts the amp.

(*) In the older discussions the idea was raised to place the elco in the HT. This can be done, but you should not cut the connecting trace but instead replace the zero-ohm resistor in this trace with a higher value to leave a DC-path for the amp to settle. That is the setup Baofeng uses.

**Reply by egzumer** [reply]
There is no 0R resistor in v1.6.
The speaker is not connected to the ground. There already is a capacitor, the one near the blob. It doesn't mitigate the pop. I know you're not actually saying that, but that kind of comes out of your description anyway, I just want to clarify.

**Reply by gartnl** [reply]
Do you mean that the external speaker is not connected to ground? I'm quit sure it is in the mike's I own. 
You're right about the 0R, there is a place for it on the 1.4 board but that has gone on the 1.6.
![R0](https://github.com/egzumer/uv-k5-firmware-custom/assets/41110951/6057acd5-135e-42cc-9d66-a4860b9cf3d0)
I expect the 'blob' capacitor to be for one of the feedback/decoupling branches. At least not in the speakerpath.

I meant the additional capacitor that is missing on the 1.6 board. If you place a capacitor there than, as far as I can see, the internal speaker would make a plop sound too, because the dc-path is missing in the bridge.

Considering the single-ended connection of the external speaker I doubt there is a hardware solution for this. Even Baofeng has the same problem ( and basic design ), and indeed the plopping sound.

This is the baofeng uv-5r. There you can see the R (1k) for the dc-adjustment over the capacitor. 
![bf design](https://github.com/egzumer/uv-k5-firmware-custom/assets/41110951/71a18f4a-8b95-4a06-8d51-cae29a36c56b)

**Reply by egzumer** [reply]
The speaker is not connected to the ground in literal sense, it is separated. There is a capacitor and parallel 47R resistor between ground and the speaker. The resistor is there to allow the ptt to function. The ptt switch pulls down the ptt line through this resistor (but it is not enough to pull it down without the extra capacitor on the other side of the speaker). The capacitor allows audio signal to drive the speaker but filters out most of the DC. It is somewchat similar to the baofeng, but the image is too small so I can't see what are the values.

**Reply by gartnl** [reply]
Ah, I know. For simplicity of explaining I just regarded it as 'ground'. For AC it can be regarded as so because the capacitor forms a short for AC. I'm sure we mean the same thing here.

**Comment by dadanielat** [comment]
Sorry for the late reply and thank you very much for the detailed comments, I can't tell my hardware revision as I actually don't want do disassemble it, but it was the cheapest UV-K5 I could find on Amazon (Jucjet brand). As I wrote it is absolutely ununseable with the headset that was shipped with the radio, but I recently got a Retevis EAK002 security headset with a 220 Ohm speaker and there is no popping sound with this one. I've also ordered a DBT-6800-K bluetooth dongle and will report my findings with this one.

![image](https://github.com/egzumer/uv-k5-firmware-custom/assets/39996336/be90d251-97f9-441c-93e8-a965b96cd2e2)
